### DISCUSSION TITLE
how to cast `cairo::FontFace` to some thing `pango::Layout` can use?

### DISCUSSION THREAD

**Discussion by ogios** [gtk-rs/gtk4-rs#1949] [Q&A]
i'm trying to load a custimized font, it's not a file but `static &[u8]`.
it's a small font and only used for numbers.

but i can only manage to load it into `cairo::FontFace` with `freetype` but don't know how to use it with `pango::Layout`:
```rust
// load into `cairo::FontFace`
let fc = ft::Library::init()
  .map_err(|e| format!("Init freetype error: {e}"))
  .and_then(|lib| {
    let fontface = lib
      .new_memory_face(Rc::new(raw::RAW_FONT.to_vec()), 0)
      .map_err(|e| format!("Init ft fontface error: {e}"))?;
    FontFace::create_from_ft(&fontface)
       .map_err(|e| format!("Init cairo fontface error: {e}"))
});

```

**Comment by sdroege** [comment]
The only way I'm aware of is by making the font available to fontconfig, and then loading it in pango via the font description.

You probably want to ask this at https://discourse.gnome.org as it's more about how to use pango in general.
